pyquil/paulis.py

```diff
@@ -82,7 +83,9 @@ def id(self):
             return s
 
     def __eq__(self, other):
-        return self.id() == other.id()
+        if not isinstance(other, PauliTerm):
+            return False
```

Comment:
We should throw a TypeError here. If two types are not comparable, there is no reason to pretend a comparison. This is also the direction that Python3 is taking.